src/main.js

```diff
@@ -336,19 +336,35 @@ class MiniGraphCard extends LitElement {
       </div>` : '';
   }
 
+  generateLegend(entity, index) {
+    let legend = this.computeName(index);
+    const state = this.getEntityState(index);
+    const { show_legend_state } = this.config.entities[index];
+
+    if (show_legend_state) {
+      legend += ` <span class="state__value">(${this.computeState(state)}</span><span class="state__uom">${this.computeUom(index)})})</span>`;
```

Comment:
Some uoms should be placed before a value - like "euro".

Reply:
Why euro should be before the unit ? I don't see it

Reply:
You probably meant "before the state".
There are no strict rules in HA about Euro, I think.
If you create a `monetary` device_class it will show UoM as for other values:
![image](https://github.com/user-attachments/assets/570c4307-d08d-4969-a8aa-4565100bcaee)
Although on Energy page it will show it differently:

![image](https://github.com/user-attachments/assets/f6e42ec0-5572-4462-b859-564f1102f284)
in fact, this "uom state" order in Energy is same for all currencies)))

Also, there was a FR: https://github.com/kalkih/mini-graph-card/issues/139
So, in future we should let users to define an order (not set a fixed order for "Euro" of course).

Reply:
I see
Maybe we should let this part for the #139 then ?

Reply:
Yes, it should be a separate PR imho. Here we will use a fixed order "state uom".